File: String Compression/string-compression.java
class Solution {
    public int compress(char[] chars) {
        int j = 0;  
        int i = 0;  
        
        while (i < chars.length) {
            char currentChar = chars[i];
            int count = 0;
            
            
            while (i < chars.length && chars[i] == currentChar) {
                i++;
                count++;
            }
            
            
            chars[j++] = currentChar;
            
            
            if (count > 1) {
                String countStr = String.valueOf(count);
                for (char c : countStr.toCharArray()) {
                    chars[j++] = c;
                }
            }
        }
        
        return j;  
    }
}

File: String Compression/string-compression.py
class Solution(object):
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        j = 0  
        i = 0 
        
        while i < len(chars):
            current_char = chars[i]
            count = 0
            
            
            while i < len(chars) and chars[i] == current_char:
                i += 1
                count += 1
            
            
            chars[j] = current_char
            j += 1
            
            
            if count > 1:
                for digit in str(count):
                    chars[j] = digit
                    j += 1
        
        return j  
